File: src/main/java/cookbook/EmailSender.java
package cookbook;

import java.util.Properties;
import java.util.Random;
import javax.mail.*;
import javax.mail.internet.*;

public class EmailSender {

    private static final String SENDER_EMAIL = "aicookbooknoreply@gmail.com"; 
    private static final String APP_PASSWORD = "mdpmgszhkczaddhf"; 

    public static String sendOTP(String recipientEmail) {
        
        String otpCode = String.format("%06d", new Random().nextInt(999999));

        Properties props = new Properties();
        props.put("mail.smtp.auth", "true");
        props.put("mail.smtp.starttls.enable", "true");
        props.put("mail.smtp.host", "smtp.gmail.com");
        props.put("mail.smtp.port", "587");
        props.put("mail.smtp.ssl.trust", "smtp.gmail.com");

        Session session = Session.getInstance(props, new javax.mail.Authenticator() {
            protected PasswordAuthentication getPasswordAuthentication() {
                return new PasswordAuthentication(SENDER_EMAIL, APP_PASSWORD);
            }
        });

        try {
            Message message = new MimeMessage(session);
            message.setFrom(new InternetAddress(SENDER_EMAIL));
            message.setRecipients(Message.RecipientType.TO, InternetAddress.parse(recipientEmail));
            message.setSubject("Your AI-CookBook Verification Code");
            
            String htmlMessage = "<html>"
            	    + "<body style='font-family: Arial, sans-serif; color: #333333;'>"
            	    + "<p>Dear " + recipientEmail + ",</p>"
            	    + "<p>Welcome to <b>Dirk's CookBook</b>!</p>"
            	    + "<p>Your verification code is:</p>"
            	    + "<h2 style='font-size: 32px; font-weight: bold; letter-spacing: 5px; color: #0e4711; margin-bottom: 10px;'>" + otpCode + "</h2>"
            	    + "<p style='font-size: 13px; color: #666666; margin-top: 0px;'>" 
            	    + "Someone tried to use this email account. If this was you, please use the code above to confirm."
            	    + "</p>"
            	    + "<p>Do not share this code with anyone. Get cooking!</p>"
            	    + "</body>"
            	    + "</html>";
            
            message.setContent(htmlMessage, "text/html; charset=utf-8");

            System.out.println("Sending email to " + recipientEmail + "...");
            
            Transport.send(message);
            
            System.out.println("Email sent successfully!");
            return otpCode; 

        } catch (MessagingException e) {
            System.out.println("Failed to send email!");
            e.printStackTrace();
            return null;
        }
    }

    public static void main(String[] args) {
        java.util.Scanner scanner = new java.util.Scanner(System.in);
        System.out.print("Enter a real email to test: ");
        String testEmail = scanner.nextLine(); 
        
        System.out.println("Starting Mail Test...");
        String generatedCode = sendOTP(testEmail);
        
        System.out.println("The code we generated and sent was: " + generatedCode);
        scanner.close();
    }
}

File: app/src/main/java/com/cookbook/backend/EmailSender.kt
package com.cookbook.backend

import com.cookbook.BuildConfig
import java.util.Properties
import java.util.Random
import javax.mail.Message
import javax.mail.PasswordAuthentication
import javax.mail.Session
import javax.mail.Transport
import javax.mail.internet.InternetAddress
import javax.mail.internet.MimeMessage

object EmailSender {

    private const val SENDER_EMAIL = "aicookbooknoreply@gmail.com"

    fun sendOTP(recipientEmail: String): String {
        val emailPassword = BuildConfig.EMAIL_PASSWORD
        if (emailPassword.isBlank()) return ""
        val otp = String.format("%06d", Random().nextInt(999999))

        val props = Properties().apply {
            put("mail.smtp.host", "smtp.gmail.com")
            put("mail.smtp.port", "587")
            put("mail.smtp.auth", "true")
            put("mail.smtp.starttls.enable", "true")
        }

        val session = Session.getInstance(props, object : javax.mail.Authenticator() {
            override fun getPasswordAuthentication(): PasswordAuthentication {
                return PasswordAuthentication(SENDER_EMAIL, emailPassword)
            }
        })

        val message = MimeMessage(session).apply {
            setFrom(InternetAddress(SENDER_EMAIL))
            setRecipient(Message.RecipientType.TO, InternetAddress(recipientEmail))
            subject = "Dirk's CookBook - Verification Code"
            setContent(
                """
                <div style="font-family: Arial, sans-serif; max-width: 500px; margin: 0 auto;">
                    <h2 style="color: #2D6A4F;">Dirk's CookBook</h2>
                    <p>Your verification code is:</p>
                    <h1 style="color: #2D6A4F; font-size: 32px; letter-spacing: 8px;">$otp</h1>
                    <p>Enter this code in the app to verify your email.</p>
                    <p style="color: #666; font-size: 12px;">
                        If you didn't request this code, please ignore this email.
                    </p>
                </div>
                """.trimIndent(),
                "text/html; charset=utf-8"
            )
        }

        Transport.send(message)
        return otp
    }
}
